pkg/kubernetes/client/apply.go

```diff
@@ -10,17 +11,17 @@ import (
 )
 
 // Apply applies the given yaml to the cluster
-func (k Kubectl) Apply(data manifest.List, opts ApplyOpts) error {
+func (k Kubectl) Apply(labels []string, data manifest.List, opts ApplyOpts) error {
```

Comment:
The labels are not required if not running with prune, right? If so, then please move them into ApplyOpts

Reply:
Not quite. The labels are needed in Apply in order to label them so that prune can work. So they are only not needed if we have `injectLabels` set to false.

Reply:
If the `k *Kubernetes` contained the environment name (or baseDir), this would reduce the need to pass it around and remove this extra parameter.

Reply:
Yup, add the whole `Environment` type in there, not just the spec :D